### PR TITLE
Nen Sack voll documentation improvements

### PR DESCRIPTION
**Pull request by funkyfuture**
a great way to procrastinate implementing important parts is to finally overhaul the docs.

mainly triggered by *autoclasstoc* being essential imo (iits functionality) and kaputt. i also finally ruled out the *sphinx-immaterial* theme which is more incomplete than one might think. and as i found *autodocsumm* as replacement for *autoclasstoc* which is seemingly coded clean enough to adapt to future *Sphinx* changes i could opt for *furo*, hossa!

this is left to do:

- [x] rebase on `main` after #80 was merged
- [x] adapt docstrings' `:meta category:`  tags, e.g. `add-nodes` -> `Adding nodes`
- [x] add a textual textual to objects that rely on obvious names and annotations so far
- [x] include `Filter`, `NodeBase`, `NodeSource` and `_TagDefinition` into the API docs with an explanatory prose, but no members
- [x] include an autodoc event handler that filters out `tuple` methods from `NamedTuple` subclasses, eg. `xpath.EvaluationContext`
- [x] investigate whether class member overviews can be ordered, if not create a feature request and take a note
- [x] figure with which commit the docstring of `utils.get_traverser` vanished, analyze the error and fix it
- ~~[ ] i'd love to create prettier tree visualizations for the Design document but haven't found anything suitable that processes text input yet~~ postponed
- [x] fix [broken hyperlinks](https://github.com/delb-xml/delb-py/actions/runs/7417426690/job/20183885874)

### CODE REVIEW

**Comment by funkyfuture** [comment]
note that a preview is best obtained from a local webserver, reading over `file:` creates issues with subsequent resource loads.

**Comment by JKatzwinkel** [comment]
shouldnt readthedocs be instructed to use a more recent python version?

File: .readthedocs.yaml [hunk 1]
```diff
@@ -3,18 +3,16 @@
...
 build:
-  image: latest
+  os: ubuntu-22.04
+  tools:
+    python: 3
 
 formats: all
 
 python:
   version: 3.7
```
Comment (funkyfuture) [inline .readthedocs.yaml:13]:
```suggestion
```

it seems that this configuration value isn't effective anymore: https://docs.readthedocs.io/en/stable/config-file/v2.html#python

File: .readthedocs.yaml [hunk 2]
```diff
@@ -3,18 +3,16 @@
 version: 2
 
 build:
-  image: latest
+  os: ubuntu-22.04
+  tools:
+    python: 3
```
Comment (funkyfuture) [inline .readthedocs.yaml:8]: @JKatzwinkel this could be more specific, but the schnöde 3 is simple enough, innit?

File: .readthedocs.yaml (hunk 1 above)
Comment (JKatzwinkel) [reply .readthedocs.yaml:13]: lmao

File: .readthedocs.yaml (hunk 2 above)
Comment (JKatzwinkel) [reply .readthedocs.yaml:8]: why not build on ubuntu-latest?

**Comment by JKatzwinkel** [comment]
i still would love to go through these properly, as it has been years since the last time i used sphinx. i'll approve regardlesz so you this doesnt collect dust while i try to get around to doing that.

**Review by JKatzwinkel — APPROVED** [review]

File: .readthedocs.yaml (hunk 2 above)
Comment (funkyfuture) [reply .readthedocs.yaml:8]: it's not available: https://docs.readthedocs.io/en/stable/config-file/v2.html#build-os

Comment (JKatzwinkel) [reply .readthedocs.yaml:8]: interesting

**Comment by funkyfuture** [comment]
i merged the changes here and new graphics for the design chapter manually.

**Pull request closed without merging**